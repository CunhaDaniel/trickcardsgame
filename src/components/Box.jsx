import React from 'react'

export default props => {
    return(
        <button onClick= {props.remount}>
            <h1>
                {props.number}
            </h1>
        </button>
    )
}