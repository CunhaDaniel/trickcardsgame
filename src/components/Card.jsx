import React from 'react';

export default props => {
  return (
    <div className="card-box">
      <img className="card" src={props.link} alt="" />

    </div>
  )
}