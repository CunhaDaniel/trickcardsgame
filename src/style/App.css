.App {
  text-align: center;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  padding: 20px;
}

.win-screen{
  display: flex;
  justify-items: center;
  align-items: center;
  flex-direction: column;
}

@media screen and (max-width: 768px){
  .App{
    padding: 10px;
    text-align: center;
    display: flex;
    justify-content: center;
    flex-direction: row;
    align-items: center;
  }
}