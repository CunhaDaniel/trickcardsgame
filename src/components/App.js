import React, { Component } from 'react';
import axios from 'axios'

import '../style/App.css';
import '../style/Pile.css'

import Card from './Card'
import Box from './Box'

class App extends Component {
  constructor(props){
    super(props);
    this.state = { cuttedDeck: this.drawDeck()}
  }

  async getDeck(){
    const deck = await axios.get('https://deckofcardsapi.com/api/deck/new/shuffle')
    .catch(err => console.log(err))
    return deck;
  }

  async remontDeck(){
    await axios.get('https://deckofcardsapi.com/api/deck/7kyhwcq3nxhi/shuffle/')
    .catch(err => console.log(err))
  }

  async drawDeck(){
    const cuttedDeck = await axios.get('https://deckofcardsapi.com/api/deck/7kyhwcq3nxhi/draw/?count=7')
    .catch(err => console.log(err))
    await this.remontDeck()
    return cuttedDeck
  }

  render() {  
    console.log(this.state.cuttedDeck)
    return (
      <div className="App">
        <div className="pile">
          <Box number={1}/>
          <Card link="https://deckofcardsapi.com/static/img/KH.png"/>
          <Card link="https://deckofcardsapi.com/static/img/KH.png" />
        </div>
        <div className="pile">
          <Box number={2}/>
          <Card link="https://deckofcardsapi.com/static/img/KH.png" />
        </div>
        <div className="pile">
          <Box number={3}/>
          <Card link="https://deckofcardsapi.com/static/img/KH.png" />
        </div>

      </div>
    );
  }
}

export default App;
