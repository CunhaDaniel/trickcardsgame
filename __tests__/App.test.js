import React from 'react'
import { shallow } from 'enzyme'

import App from '../src/components/App'
import Box from '../src/components/Box'
import Card from '../src/components/Card'

describe('Testing component box', () => {
  it('should render correctly box component', () => {
    const wrapper = shallow(
      <Box number={5} />
    )
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ number: 8 })
    expect(wrapper).toMatchSnapshot();
  })
})


describe('Testing component app', () => {
  it('should render correctly app component', () => {
    const wrapper = shallow(
      <App />
    )
    expect(wrapper).toMatchSnapshot();
  })
})

describe('Testing component card', () => {
  it('should render correctly card component', () => {
    const wrapper = shallow(
      <Card  link={"https://deckofcardsapi.com/static/img/6S.png"}/>
    )
    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({ link: "https://deckofcardsapi.com/static/img/8H.png" })
  })
})