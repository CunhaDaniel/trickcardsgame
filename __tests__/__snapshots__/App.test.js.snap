// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component app should render correctly app component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="pile"
        >
          <_default
            number={1}
            remount={[Function]}
          />
        </div>,
        <div
          className="pile"
        >
          <_default
            number={2}
            remount={[Function]}
          />
        </div>,
        <div
          className="pile"
        >
          <_default
            number={3}
            remount={[Function]}
          />
        </div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <_default
              number={1}
              remount={[Function]}
            />,
            Array [],
          ],
          "className": "pile",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "number": 1,
              "remount": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <_default
              number={2}
              remount={[Function]}
            />,
            Array [],
          ],
          "className": "pile",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "number": 2,
              "remount": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <_default
              number={3}
              remount={[Function]}
            />,
            Array [],
          ],
          "className": "pile",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "number": 3,
              "remount": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="pile"
          >
            <_default
              number={1}
              remount={[Function]}
            />
          </div>,
          <div
            className="pile"
          >
            <_default
              number={2}
              remount={[Function]}
            />
          </div>,
          <div
            className="pile"
          >
            <_default
              number={3}
              remount={[Function]}
            />
          </div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <_default
                number={1}
                remount={[Function]}
              />,
              Array [],
            ],
            "className": "pile",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "number": 1,
                "remount": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <_default
                number={2}
                remount={[Function]}
              />,
              Array [],
            ],
            "className": "pile",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "number": 2,
                "remount": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <_default
                number={3}
                remount={[Function]}
              />,
              Array [],
            ],
            "className": "pile",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "number": 3,
                "remount": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing component box should render correctly box component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    number={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        5
      </h1>,
      "className": "btn",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 5,
      },
      "ref": null,
      "rendered": 5,
      "type": "h1",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          5
        </h1>,
        "className": "btn",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 5,
        },
        "ref": null,
        "rendered": 5,
        "type": "h1",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing component box should render correctly box component 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    number={8}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        8
      </h1>,
      "className": "btn",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": 8,
      },
      "ref": null,
      "rendered": 8,
      "type": "h1",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          8
        </h1>,
        "className": "btn",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 8,
        },
        "ref": null,
        "rendered": 8,
        "type": "h1",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing component card should render correctly card component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    link="https://deckofcardsapi.com/static/img/6S.png"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt=""
        className="card"
        src="https://deckofcardsapi.com/static/img/6S.png"
      />,
      "className": "card-box",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "",
        "className": "card",
        "src": "https://deckofcardsapi.com/static/img/6S.png",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt=""
          className="card"
          src="https://deckofcardsapi.com/static/img/6S.png"
        />,
        "className": "card-box",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "card",
          "src": "https://deckofcardsapi.com/static/img/6S.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
